pandas/core/indexes/multi.py

```diff
@@ -1774,14 +1784,20 @@ def to_frame(self, index: bool = True, name=lib.no_default) -> DataFrame:
         else:
             idx_names = self.names
 
+        idx_names = [
```

Comment:
umm why are you repeating L1785?

Reply:
It is doing a transform: filling in None names with the level number.

Whether that is the right thing to do is another issue. I am just preserving the existing behavior,

Reply:
then this needs another argument similar to how this is done in `.reset_index`

Reply:
I am not changing anything here. The old code is https://github.com/johnzangwill/pandas/blob/6cc5584bba59ef8f06d4dc901dc39ddd08d1519f/pandas/core/indexes/multi.py#L1780:

`
                (level if lvlname is None else lvlname): self._get_level_values(level)
`

and I have just moved that logic earlier, since I need unique dictionary indexes.

In any case, insert and reset_index do this differently, replacing None level labels with level_n. As I say, that is a separate issue and I have raised it elsewhere (#45245), but is is not the subject of this PR, 

I don't think that this is conditional in reset_index or that there is an argument for it. Which argument are you referring to?

This is the code in reset_index:

```
            if isinstance(self.index, MultiIndex):
                names = com.fill_missing_names(self.index.names)
                to_insert = zip(self.index.levels, self.index.codes)
            else:
                default = "index" if "index" not in self else "level_0"
                names = [default] if self.index.name is None else [self.index.name]
                to_insert = ((self.index, None),)
```

that puts in "level_n" for multi-index and "index" or "level_0" for simple index.

Reply:
ok can you just make a method on Index then to do this, repeating this code is not great

Reply:
I have searched Pandas and I cannot find any other instance of this. The nearest is https://github.com/pandas-dev/pandas/blob/4e034ec0006b6c05160ce67ea1420ce28f295c91/pandas/core/indexes/base.py#L1631 which does implement the policy (on self.name, not self.names). I can factor that down if you think that it is worth it.

Reply:
Ok, but I have already explained many times that `reset_index` does _not_ do this.

`reset_index`, `to_records` and many other methods all fill the None entries  with "level_n", not with "n". As you know, I factored those out into a common method (`com.fill_missing_names` #44878) which is invoked in 6 different places.

These `MI/Index.to_frame` methods are the _only_ ones which do it differently, filling the gaps with the column number. This difference could be discussed, and I have made an issue (#45245), but I don't suggest changing it without a lot of thought. Changing `to_frame` would break virtually all its tests.